### ISSUE TITLE
Rendering has been broken?

### ISSUE THREAD

**Issue by brambox** [whscullin/apple2js#85]
I used an old version of the emulator and it had superior crispness and colors.
Now it's blurry and over brighten full disaster. I see this new GL thing but even after turning it off and reload it only gets a bit better but it's still no way near to the old rendering. Can you please fix it to have a proper crispness and RGB colors? 
Maybe there is a temporary fix I can set somewhere in the code to use the old rendering way.
Thank you!

**Comment by whscullin** [comment]
Non-NTSC shader rendering is largely unchanged, did you reload the page after changing the preference as mentioned in the `*`?

**Comment by brambox** [comment]
Yes. I will make pic. https://imgur.com/a/YVD1daO 
top is the new

**Comment by whscullin** [comment]
Ah. Yeah, that predates the introduction of the GL renderer by a bit. Somebody pointed out that I was using the IIGS colors instead of NTSC correct colors, so I corrected that. The difficulty of trying to please everybody, I guess. 🤷‍♂️

**Comment by brambox** [comment]
It's not just the colors change. The main thing is it is blurry like some kind of filter applied.
Here on this pic it is easy to see on the Apple Text. https://imgur.com/L1rzpQL
Top again is new and if you enable GL it's even worse.
Is it possible to fix it?

**Comment by iflan** [comment]
Hi there,

GL mode is trying to actually simulate what you'd see if you were looking at an old-school CRT monitor, including barrel effects, the shadow mask, and so on. If you're looking for crisp graphics, then GL mode is not what you want.

In regular mode, there are still rendering artifacts due to how the Apple actually drew hi-res graphics. You can read about it a bit here:

https://en.wikipedia.org/wiki/Apple_II_graphics#High-Resolution_.28Hi-Res.29_graphics

There's a similar issue that was opened on AppleWin:

https://github.com/AppleWin/AppleWin/issues/357

In the end, there's a tug of war between accurate rendering and "pretty" rendering. I don't know if there's any "best" solution except having several choices. But giving people all the choices also means maintaining all those choices as well.

Ian

**Comment by whscullin** [comment]
I've updated the non-GL canvas rendering to use an anti-smoothing flag I uncovered when doing some work on full page rendering. Chrome seems to respect it, Safari and Firefox do their own thing, of course, but they were doing something different before, too.

**Comment by brambox** [comment]
Thank you! 
Will it be possible to get the rgb card rendering like with the gl option on but without the bluring?

**Comment by whscullin** [comment]
I'm not quite sure what you mean. The eventual goal behind keeping two renderers is to have one renderer that emulates NTSC and its artifacts, the other to emulate something like the Video 7/Apple Extended RGB card, with no artifact colors and sharp B&W modes.

I do plan on making the NTSC mode more tunable, the underlying library has very flexible control over the level of artifacting, it's just a little tricky fitting that many dials into the UI.
